src/main/java/duke/Commands.java

```diff
@@ -270,4 +270,28 @@ public void delete(int index, DukeHistory history)  {
         }
         history.deleteTask(index);
     }
+
+    /**
+     * A method that, when called, builds a String phrase with the inputted String[] tokens before getting the
+     * inputted instance of DukeHistory to call findPhrase().
+     *
+     * It then prints the returned String for the user to see.
+     *
+     * @param tokens A String[] of tokens inputted by the user.
+     * @param history An instance of DukeHistory.
+     * @throws DukeException If no task is found matching the inputted phrase.
+     */
+    public void find(String[] tokens, DukeHistory history) throws DukeException {
+        StringBuilder phrase = new StringBuilder();
+        for (int i = 1; i < tokens.length; i++) {
+            phrase.append(tokens[i]);
+            if (i != (tokens.length - 1)) {
+                phrase.append(" ");
+            }
+        }
+        String border = "_______________________________________________________\n";
+        System.out.println(border
+                + "Here is what we found:\n"
+                + history.findPhrase(phrase.toString()) + border);
+    }
```

Comment:
This command class may be a little too long?
It would be great if you could split up the find and delete command, and extend them from a command class. It will be easier to sort and debug if you encounter any issues:)
an example could be `public class FindCommand extends Command {}`